openmc/deplete/abc.py

```diff
@@ -759,8 +759,13 @@ def _get_bos_data_from_restart(self, step_index, source_rate, bos_conc):
     def _get_start_data(self):
         if self.operator.prev_res is None:
             return 0.0, 0
-        return (self.operator.prev_res[-1].time[-1],
-                len(self.operator.prev_res) - 1)
+        else:
+            if comm.size != 1:
+                return (self.operator.prev_res[-1].time[-1],
+                        int(len(self.operator.prev_res)/2) - 1)
+            else:
+                return (self.operator.prev_res[-1].time[-1],
+                        len(self.operator.prev_res) - 1)
```

Comment:
I agree with @gridley, please add comments to this potion of code explaining why this is necessary.